### ISSUE TITLE
Case has not the correct indentation in a switch

### ISSUE THREAD

**Issue by gimler** [FriendsOfPHP/PHP-CS-Fixer#776]
Here is a failing testcase for PSR2 BracesFixerTest add to provideFixWhitespaceBeforeBraceCases

``` php
        return array(
            array(
                '<?php
    switch (n) {
        case label1:
            echo 1;
            echo 2;
            break;
        default:
            echo 3;
            echo 4;
    }',
                '<?php
    switch (n)
    {
     case label1:
        echo 1;
            echo 2;
            break;
        default:
            echo 3;
            echo 4;
    }',
            ),
```

version master

``` php
$finder = Symfony\CS\Finder\DefaultFinder::create()
    ->in('src')
    ->files()
    ->name('*.php')
    ->ignoreDotFiles(true)
    ->ignoreVCS(true)
    ->exclude('vendor')
;

return Symfony\CS\Config\Config::create()
    ->level(Symfony\CS\FixerInterface::PSR2_LEVEL)
    ->fixers(array('unused_use', 'duplicate_semicolon'))
    ->finder($finder)
;
```

**Comment by chrisRidgers** [comment]
Does anyone have an update on this?  One of the issues referencing this was closed and declared not current (fair enough), but what is the current state of switch/ case blocks in php-cs-fixer?

**Comment by nicolas-grekas** [comment]
An other failing test case:
before:
```
switch ($foo)
{
case 'bar': if (5) echo 6;
}
```

after:
```
switch ($foo) {
case 'bar': if (5) {
     echo 6;
 }
}
```

**Comment by MortimerCat** [comment]
PHP-cs-Fixer now seems to ignore all content within a switch statement
```
<?php

class mySillyClass
{
public function mrMethod() {
switch ($i) {
case 0:
echo "i equals 0";
break;
case 1:
echo "i equals 1";
break;
case 2:
echo "i equals 2";
break;
}
}
} 
```
becomes 
```
<?php

class mySillyClass
{
    public function mrMethod()
    {
        switch ($i) {
case 0:
echo "i equals 0";
break;
case 1:
echo "i equals 1";
break;
case 2:
echo "i equals 2";
break;
}
    }
}
```
or am I expecting too much?

**Comment by keradus** [comment]
If I remember correctly there were never support of adding indent to body of `case` (willing to add? go ahead! :+1: for it), yet no clue out of the box why case is not indented for `{` :/ I would say it shall do it, but don't remember was it ever able to. Are you willing to take a deeper look and maybe send a PR?

**Comment by sgehrman** [comment]
same problem, doesn't work

**Comment by keradus** [comment]
@sgehrman 
let me repeat the essential of this thread.
Requested feature was never implemented, so obviously it cannot work.
If you are interested in it, please give a helpful hand and send PR.

**Comment by sstok** [comment]
To help the one who is going to implement this be sure to follow the http://www.php-fig.org/psr/psr-2/#52-switch-case

I guess https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.10/src/Fixer/Basic/BracesFixer.php needs to be updated? but are there any special options that are concerned with this? 🤔

And is any of the current fix re-usable 😄 I can try to make it work, but I haven't contributed to this project in years! And want to avoid re-doing work 😝

**Comment by keradus** [comment]
Modify `BracesFixer` is one of the way to do it indeed, yet, maybe it is better to actually do it as separated fixer instead, to not extend this little monster even further.

https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.10/src/Fixer/Basic/BracesFixer.php#L292 
this one is extracting `switch` from being fixed. Thing is that `braces` rule is taking care about... `braces`. And as long as `switch` structure has braces, then `case` usually not, which means that calculating of proper indentation that is there in `BracesFixer` will not work properly for such case - this is exactly the reason why it was not handled in current code.

**Comment by kisanme** [comment]
Any update or fix on this?

**Comment by ntzm** [comment]
Nobody has proposed a PR yet, if you'd like to have a go that would be great!

**Comment by SpacePossum** [comment]
related https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2650

**Comment by muuvmuuv** [comment]
Can we not transport the function that indents switch statements from PHPCS? I think they got what we need here.

And +1 to add these feature. I poorly don't have the skill in PHP to add these by myself, so I hope someone will.

**Comment by keradus** [comment]
PHPCS is way different software used totally different solution in background (way of dealing with tokenization)

**Comment by bskrtich** [comment]
I have posted a $20 bounty for this issue to be fixed.

https://www.bountysource.com/issues/6600347-case-has-not-the-correct-indentation-in-a-switch

**Comment by julienfalque** [comment]
Fixed in #4884.
